packages/workflow/src/internals.ts

```diff
@@ -577,6 +686,29 @@ export class Activator implements ActivationHandler {
     }
   }
 
+  public dispatchBufferedUpdates(): void {
+    const bufferedUpdates = this.bufferedUpdates;
+    while (bufferedUpdates.length) {
+      const foundIndex = bufferedUpdates.findIndex((update) => update.name && this.updateHandlers.has(update.name));
+      if (foundIndex === -1) {
+        // No buffered Updates have a handler yet.
+        break;
+      }
+      const [update] = bufferedUpdates.splice(foundIndex, 1);
+      this.doUpdate(update);
+    }
+  }
+
+  public rejectBufferedUpdates(): void {
+    while (this.bufferedUpdates.length) {
+      const [update] = this.bufferedUpdates.splice(0, 1);
```

Comment:
nit: Use `shift()` instead of `slice()` if you simply want the first item.

Reply:
Thanks! f4a3126394f35557327f747d06ca05ff9ba09ecb